arrow-libs/core/arrow-core/src/main/kotlin/arrow/core/Ior.kt

```diff
@@ -505,6 +514,9 @@ sealed class Ior<out A, out B> {
       { a, b -> fa(b).map { Both(a, it) } }
     )
 
+  inline fun <C> traverseOption(fa: (B) -> Option<C>): Option<Ior<A, C>> =
+    traverseEither { fa(it).toEither { Unit } }.orNone()
```

Comment:
```suggestion
    fold(
      { a -> Some(Left(a)) },
      { b -> fa(b).map { Right(it) } },
      { a, b -> fa(b).map { Both(a, it) } }
    )
```

Nit: a direct implementation reads better IMO, and is (slightly) more performant.